crates/evm/src/lib.rs

```diff
@@ -118,7 +71,11 @@ pub trait ConfigureEvm: ConfigureEvmEnv {
     /// # Caution
     ///
     /// This does not initialize the tx environment.
-    fn evm_for_block<DB: Database>(&self, db: DB, header: &Self::Header) -> Self::Evm<'_, DB, ()> {
+    fn evm_for_block<DB: Database>(
+        &self,
+        db: DB,
+        header: &Self::Header,
+    ) -> <Self::EvmFactory as EvmFactory<EvmEnv<Self::Spec>>>::Evm<'_, DB, ()> {
```

Comment:
could be helpful to create a type alias as adapter for this and export it from crate root to improve devx with the trait, smthg like
```rust
pub type EvmTy<'a, T, DB> = <<T as ConfigureEvm>::EvmFactory<EvmEnv<T::Spec>>>::Evm<'a, DB, ()>;
```